src/Resources/AbstractResource.php

```diff
@@ -8,11 +8,34 @@ abstract class AbstractResource
 {
     public function toArray(): array
     {
-        return (array) $this;
+        $array = [];
+
+        $this->__objectToArray($this, $array);
+
+        return $array;
     }
 
     public function toJson(): string
     {
         return json_encode($this);
     }
+
+    private function __objectToArray($obj, &$arr)
```

Comment:
Why underscores? That's not common syntax. And why mutate the input?

Seems like lines 11 & 15 are unnecessary and this could just be the body of __objectToArray